src/components/ItemProvider/ItemProvider.tsx

```diff
@@ -1,18 +1,26 @@
 import * as React from 'react'
-import { Props } from './ItemProvider.types'
+import { Props, State } from './ItemProvider.types'
+
+export default class ItemProvider extends React.PureComponent<Props, State> {
+  state: State = {
+    loadedItemId: undefined
+  }
 
-export default class ItemProvider extends React.PureComponent<Props> {
   componentDidMount() {
     const { id, item, onFetchItem, isConnected } = this.props
-    if (isConnected && id && !item) {
-      onFetchItem(id)
+    const { loadedItemId } = this.state
+
+    if (isConnected && id && !item && loadedItemId !== id) {
```

Comment:
Do we need to check `loadedItemId !== id` in the `DidMount`? I believe that  `loadedItemId` will be `undefined` here.